### PR TITLE
first stab at writing tests for writing tests for async in F#

### PR DESCRIPTION
**Pull request by haf**
Related to #34 - writing async F# tests

### CODE REVIEW

File: src/framework/nunit.framework-4.5.csproj [hunk 1]
```diff
@@ -369,6 +369,10 @@
     <Compile Include="Constraints\ExceptionTypeConstraint.cs" />
   </ItemGroup>
   <ItemGroup>
+    <Reference Include="FSharp.Core, Version=4.3.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
```
Comment (rprouse) [inline src/framework/nunit.framework-4.5.csproj]: We shouldn't be pulling language specific assemblies into the framework. There is no guarantee consumers of NUnit even have F# installed. I assume this doesn't ship with the core framework based on the hintpath?

File: src/framework/Internal/Commands/TestMethodCommand.cs [hunk 2]
```diff
@@ -74,14 +76,43 @@ public override TestResult Execute(TestExecutionContext context)
...
+        {
+            object res = Reflect.InvokeMethod(testMethod.Method, context.TestObject, arguments);
+
+            try
+            {
+                var parameters = new[]
+                    {
+                        res,
+                        context.TestCaseTimeout > 0
+                            ? Microsoft.FSharp.Core.FSharpOption<int>.Some(context.TestCaseTimeout)
```
Comment (rprouse) [inline src/framework/Internal/Commands/TestMethodCommand.cs]: Based on my comment below, I don't think we should be including language specific code in the framework.

**Comment by rprouse** [comment]
This pull request is very out of date and there has been no movement on it in over a year. Shall we close?

**Comment by CharliePoole** [comment]
I agree. I asked for some info over a year ago and there has been no response from @haf since then.

**Comment by haf** [comment]
I think you should close it, actually. If someone wants to come and perform the needed extra work, then I would help guide them, but right now I don't have the needed time and/or momentum to finish this work. Sorry and thank you for your input on what was done.

**Pull request closed without merging**